### DISCUSSION TITLE
Continuous breaking changes

### DISCUSSION THREAD

**Discussion by hotrush** [riverqueue/river#694] [General]
Hello, dear maintainer.

We like your package and appreciate your job, but continuous breaking changes in minor and patch releases bring bad developer experience:

- broken CI pipelines when using any of auto-updating software 
- potentially broken productions
- it's just a bad practice 

We had issues like that at least twice in last 6 months.

When breaking changes need to be made it makes more sense to collect them and release a new major release, it is safe and common way of doing this.

**Comment by bgentry** [comment]
Hi, can you elaborate on what specific changes you were impacted by? Was it the removal of the old unique job design in v0.13.0, some specific database migrations, or something older than that? Or are you seeing CI breakages because of the need to upgrade River and riverui at the same time? Want to make sure we’re thinking about the right problems.

We aren’t yet at a v1.0, and while we’ve tried hard to minimize breaking changes some have been necessary in order to get to what we think is a stable foundation and base set of functionality. In some cases certain primitives were just wrong in their initial design (error handler breaking changes for example) and needed fixing.

Once we release a v1 we won’t have breaking API changes without another major version, though we may still limit new versions to supporting to the most recent few years of Go and Postgres releases.

**Reply by hotrush** [reply]
Yeah, I expected answer like this. I hope you'll tag a major release soon 😄  But still, I think many people use it in production...

We were affected by 0.13 and 0.12, didn't use older versions.
